weldx/transformations/local_cs.py

```diff
@@ -310,6 +310,13 @@ def _build_coordinates(
                     extra_msg="\nThe coordinates require units representing a length.",
                 )
 
+        if not isinstance(coordinates.data, pint.Quantity):
+            warnings.warn(
+                "Coordinates without units are deprecated and won't be supported in "
+                "the future",
+                DeprecationWarning,
```

Comment:
please add a stacklevel argument here, to inform the callee of this method. I guess 2 in this case, since it is a private method.